### PR TITLE
Modify HAL to use OV 2.0 API's for graph creation and inference

### PR DESCRIPTION
**Pull request by Rjasuja**
Update graph creation as per OV 2.0 API's
Modify namespace from ngraph to ov

Signed-off-by: Ritul Jasuja <ritul.jasuja@intel.com>

### CODE REVIEW

File: BasePreparedModel.h [hunk 1]
```diff
@@ -43,20 +44,13 @@ using ::android::hardware::MQDescriptorSync;
...
-namespace hardware {
-namespace neuralnetworks {
-namespace nnhal {
-
+namespace android::hardware::neuralnetworks::nnhal {
 template <class T>
 using vec = std::vector<T>;
 typedef uint8_t* memory;
 
 class BasePreparedModel : public V1_3::IPreparedModel {
```
Comment (RavirajSitaram) [inline BasePreparedModel.h:52]: Rename BasePreapredModel to IntelDevicePreparedModel??

File: cpu/CpuPreparedModel.cpp [hunk 2]
```diff
@@ -35,20 +35,20 @@ bool CpuPreparedModel::initialize() {
 
     if (!mNgraphNetCreator->validateOperations()) return false;
     ALOGI("Generating IR Graph");
-    auto ngraph_function = mNgraphNetCreator->generateGraph();
```
Comment (RavirajSitaram) [inline cpu/CpuPreparedModel.cpp:38]: Delete CpuPreapredModel as it is no longer needed?

File: gna/GnaPreparedModel.cpp [hunk 3]
```diff
@@ -46,7 +46,7 @@ bool GnaPreparedModel::initialize() {
 #else
     ngraph_net->serialize("/tmp/ngraph_ir.xml", "/tmp/ngraph_ir.bin");
 #endif
```
Comment (RavirajSitaram) [inline gna/GnaPreparedModel.cpp:48]: Delete GnaPreparedModel?

**Review by RavirajSitaram — COMMENTED** [review]
Changes look good. If not in this PR, consider the review comments in future PR's

**Comment by RavirajSitaram** [comment]
@Rjasuja  Involve reviewer from Android team for this change.

**Comment by Rjasuja** [comment]
I am unable to add reviewer from android team. Can somebody else try to add who has access

**Comment by anikulk** [comment]
@reaganlo looks like some error in ci automation, can you check.

**Pull request merged**